src/theme/components/template.ts

```diff
@@ -1,103 +1,75 @@
 import { defineStyleConfig } from "@chakra-ui/react";
 import { defineStyle } from "@chakra-ui/system";
-/**
- * Grid layout based on https://www.joshwcomeau.com/css/full-bleed/
- */
+
+export const responsiveGap = { base: "1rem", md: "1.5rem", xl: "1rem" };
+export const responsivePadding = { base: "1rem", md: "1.5rem", xl: "2rem" };
 
 const Template = defineStyleConfig({
   baseStyle: defineStyle({
     boxSizing: "border-box",
     color: "ui.typography.body",
-    display: "grid",
-    gridTemplateColumns: `
-      1fr
-      min(1280px, 100%)
-      1fr`,
-    rowGap: "grid.l",
     _dark: {
       color: "dark.ui.typography.body",
     },
   }),
   sizes: {},
   defaultProps: {},
 });
-// Elements that need to breakout will span outside
-// the center 1280px grid column.
-const TemplateBreakout = defineStyleConfig({
-  baseStyle: defineStyle({
-    width: "100%",
-    // This could be "1 / 4" and it would mean the same. This is
-    // "future-proof" the grid column assignment to the last column.
-    gridColumn: "1 / -1",
-  }),
-});
+
 const TemplateContent = defineStyleConfig({
   baseStyle: defineStyle({
-    // Set this element to start on the second 1280px grid column.
-    gridColumn: "2",
-    // This element also contains its own grid system within, but we use "flex"
-    // for mobile to deal with overflow issues related to the Table component.
-    display: { base: "flex", md: "grid" },
-    flexDirection: { base: "column", md: null },
-    gridTemplateColumns: "1fr",
-    paddingY: 0,
-    paddingX: "s",
-    gap: "grid.l",
+    display: "grid",
+    gridTemplateColumns: {
+      base: "repeat(1, minmax(100px, 1fr))",
+    },
+    gap: responsiveGap,
+    padding: responsivePadding,
   }),
-  // With left or right sidebars, we need to set two grid columns and
-  // the column for the sidebar is max 255px width.
+
   variants: {
+    /** If neither `left` nor `right` variants are used, we default to `none` and
+     * the grid will show 1 column only at any viewport width. */
     left: {
-      gridTemplateColumns: { md: "255px 1fr" },
+      gridTemplateColumns: {
+        md: "repeat(2, minmax(100px, 1fr))",
+        lg: "1fr 2fr",
+        xl: "1fr 3fr",
+      },
     },
     right: {
-      gridTemplateColumns: { md: "1fr 255px" },
+      gridTemplateColumns: {
+        md: "repeat(2, minmax(100px, 1fr))",
+        lg: "2fr 1fr",
+        xl: "3fr 1fr",
+      },
```

Comment:
This is so much more straight forward than what I put together in my POC using all 12 columns. Easier on the eyes and brain. However, I am still curious about the widths of the columns and gaps and how they align with expectations. Once the `responsiveGap` and `responsiveMargin` variables are updated and implemented, I will compare the widths in this POC with the widths in my POC. If they are in alignment, then I don't know what I was seeing that caused me to implement a full 12-column grid.